File: problems/python/1/Solution1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numsToIdx = dict()
        for i in range(0, len(nums)):
            complement = target - nums[i]
            if complement in numsToIdx:
                return [numsToIdx[complement], i]
            numsToIdx[nums[i]] = i
        return []

File: problems/kotlin/1/Solution1.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val numsToIdx = HashMap<Int, Int>()
        for ((idx, num) in nums.withIndex()) {
            val complement = target - num
            numsToIdx[complement]?.let { return intArrayOf(it, idx) }
            numsToIdx[num] = idx
        }
        return intArrayOf()
    }
}